Mutagen.Bethesda.Analyzers.Api/IAnalyzer.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System.IO.Abstractions;
+using Autofac;
+using Mutagen.Bethesda.Analyzers.Autofac;
+using Mutagen.Bethesda.Analyzers.Config.Run;
+using Mutagen.Bethesda.Analyzers.Config.Topic;
+using Mutagen.Bethesda.Analyzers.Drivers;
+using Mutagen.Bethesda.Analyzers.Reporting.Drops;
+using Mutagen.Bethesda.Analyzers.Reporting.Handlers;
+using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Drops;
+using Mutagen.Bethesda.Analyzers.Skyrim;
+using Mutagen.Bethesda.Environments;
+using Mutagen.Bethesda.Environments.DI;
+using Mutagen.Bethesda.Plugins;
+using Mutagen.Bethesda.Plugins.Meta;
+using Mutagen.Bethesda.Plugins.Records;
+using Noggog.WorkEngine;
+
+namespace Mutagen.Bethesda.Analyzers.Api;
+
+public static class AnalyzerExtensions
+{
+    public static IAnalyzer Analyzer(this IGameEnvironment gameEnvironment, AnalyzerOptions? options = null)
+    {
+        return new Analyzer(gameEnvironment, options);
+    }
+}
+
+public interface IAnalyzer
+{
+    /// <summary>
+    /// Analyze a mod for issues.
+    /// </summary>
+    /// <param name="mod">Mod to analyze</param>
+    /// <returns>Analysis results for topics found in the mod </returns>
+    Task<IEnumerable<AnalyzerResult>> Analyze(IModGetter mod);
+
+    /// <summary>
+    /// Analyze a major record for issues.
+    /// </summary>
+    /// <param name="record">Major record to analyze</param>
+    /// <returns>Analysis results for topics found in the record</returns>
+    Task<IEnumerable<AnalyzerResult>> Analyze<TMajorRecord>(TMajorRecord record)
+        where TMajorRecord : IMajorRecordGetter;
+}
+
+public class Analyzer : IAnalyzer
+{
+    private readonly IGameEnvironment _gameEnvironment;
+    private readonly IWorkDropoff _workDropoff;
+    private readonly IContainer _container;
+
+    public IDriverProvider<IContextualDriver> ContextualModDrivers { get; }
+    public IDriverProvider<IIsolatedDriver> IsolatedModDrivers { get; }
+
+    public Analyzer(IGameEnvironment gameEnvironment, AnalyzerOptions? options = null)
```

Comment:
I think the ctor itself should be adjusted a bit.
- Inject IFileSystem into the ctor, rather than assuming a real filesystem
- IGameEnvironment is too constricting.  It needs a few things GameEnvironment might provide, but it doesn't actually need all that.  From the game environment, it needs:
   - GameRelease
   - LoadOrder
   - LinkCache

The helper functions going from the GameEnvironment to an analyzer can get the appropriate things and pass them along, rather than fundamentally requiring a GameEnvironment

On that note, i think we can provide static Factory functions on this class to expose the alternatives conveniently.   One that takes in those parameters exactly, one that takes in a game environment, maybe some other flavors.